### PR TITLE
Centralized exception throwing if builder is turned into a FormConfig…

### PR DESCRIPTION
**Pull request by unckleg**
| Q             | A
| ------------- | ---
| Branch?       | 3.4 
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes/no
| Fixed tickets |  #...
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Thanks for submitting this PR.
Still, I'm :-1::
- the method only pushed the exceptions one level higher on the stack, and makes changes their file+line info
- the annotation documents a abnormal case
- more importantly, this is pure code style change: the existing code is perfectly fine as is

**Comment by unckleg** [comment]
This is my first PR :)

- I thought it is valid following DRY principle.
- Abnormal or normal it is again the case that it can come to and Exception throwing should be considered as part of method annotation. 
- Are CS changes being considered at all ?

**Comment by nicolas-grekas** [comment]
> This is my first PR

I saw that, congrats :) For your next PRs (I hope there will be), you need to target the lowest applicable branch (2.7 here I'd say.)

> following DRY principle

DRY doesn't mean "factorize lines of code" but "factorize semantic units". Here, whether this "if+exception" is a semantic unit or not is debatable. I'd say no myself.

> method annotation

for sure, the annotation should be on the implementation, not on the class - this behavior is implementation specific. But then what value do the annotation add? Do you think people will be more efficient after they've been added? I really doubt - but I'm sure this adds "visual debt" ;)

> Are CS changes being considered at all ?

yes, but very prudently - and we dismiss most of them unless they provide value to the maintenance process...

File: src/Symfony/Component/Form/FormBuilder.php [hunk 1]
```diff
@@ -271,4 +253,11 @@ private function resolveChildren()
 
         $this->unresolvedChildren = array();
     }
+
+    protected function throwExceptionIfLocked()
```
Comment (iltar) [inline src/Symfony/Component/Form/FormBuilder.php]: Should probably be private, no reason to have it protected.

**Comment by nicolas-grekas** [comment]
Shall we close as explained?

**Comment by ogizanagi** [comment]
I agree with @nicolas-grekas here. Let's close this.

@unckleg : Anyway, thanks a lot for taking the time to contribute. I hope it won't discourage you to submit new PRs :)

**Comment by unckleg** [comment]
Thanks for closing @ogizanagi!
I hope sending some PR again very soon :D

**Pull request closed without merging**